### ISSUE TITLE
ActiveJob named parameters

### ISSUE THREAD

**Issue by radeno** [moove-it/sidekiq-scheduler#90]
Hello,

i try to figure-out how to pass ```args``` to ActiveJob class

for example:

```ruby
Sidekiq.set_schedule('job_name', 'SomeJob', cron: '* * * * *', args: { message_id: 23121, regenerate: true } )
```
but in method ```enqueue_job``` is 
```ruby
      if defined?(ActiveJob::Enqueuing) && config['class'].included_modules.include?(ActiveJob::Enqueuing)
        config['class'].new.enqueue(config)
      else
```
```ActiveJob::Enqueuing.enqueue``` doesn't accept job arguments but `initialize` does.

Passing config['args'] to initialize works:
Just for example:
```ruby
    def self.enqueue_job(job_config)
      config = job_config.dup

      config['class'] = config['class'].constantize if config['class'].is_a?(String)
      if config['args'].is_a?(Hash)
        config['args'].symbolize_keys!
      else
        config['args'] = Array(config['args'])
      end

      if defined?(ActiveJob::Enqueuing) && config['class'].included_modules.include?(ActiveJob::Enqueuing)
        config['class'].new(config['args']).enqueue
      else
        Sidekiq::Client.push(config)
      end
    end
```

**Comment by andrewhavens** [comment]
I think I am running into this issue also. I just switched to the new "active job supported configuration syntax" that was added in #80. The admin UI displays that it will pass arguments, but I get errors when the jobs run about missing arguments.

**Comment by karmacoma** [comment]
I also have this same problem. Arguments are not getting passed.

**Comment by radeno** [comment]
My temporary monkey patch (put into ``lib`` directory)

```ruby
module Sidekiq
  class Scheduler
    # Enqueue a job based on a config hash
    def self.enqueue_job(job_config)
      config = job_config.dup

      config['class'] = config['class'].constantize if config['class'].is_a?(String)
      if config['args'].is_a?(Hash)
        config['args'].symbolize_keys!
      else
        config['args'] = Array(config['args'])
      end

      if defined?(ActiveJob::Enqueuing) && config['class'].included_modules.include?(ActiveJob::Enqueuing)
        config['class'].new(config['args']).enqueue
      else
        Sidekiq::Client.push(config)
      end
    end
  end
end
```

**Comment by karmacoma** [comment]
@radeno: Thank you! :+1: I will use this for the time being.

**Comment by elpic** [comment]
Please take a look at the enqueue method it does take some arguments

http://api.rubyonrails.org/classes/ActiveJob/Enqueuing.html#method-i-enqueue

I'm currently working on a fix for this

**Comment by andrewhavens** [comment]
Well, I figured out why my arguments weren't being passed to my job...it was because I wasn't specifying the args as an array:

```yml
# sidekiq.yml

# before (not working):
args: "just a string"

# after (working):
args:
  - "just a string"
```

@radeno I'm guessing it will work if you pass your hash of args as a single item in an array, like this:

```ruby
Sidekiq.set_schedule('job_name', 'SomeJob', cron: '* * * * *', args: [{ message_id: 23121, regenerate: true }])
```

**Comment by elpic** [comment]
Already solved on v2.0.5
